Userland/Libraries/LibJS/Runtime/ObjectConstructor.cpp

```diff
@@ -182,6 +184,40 @@ JS_DEFINE_NATIVE_FUNCTION(ObjectConstructor::freeze)
     return argument;
 }
 
+// 20.1.2.7 Object.fromEntries ( iterable ), https://tc39.es/ecma262/#sec-object.fromentries
+JS_DEFINE_NATIVE_FUNCTION(ObjectConstructor::from_entries)
+{
+    auto iterable = require_object_coercible(global_object, vm.argument(0));
+    if (vm.exception())
+        return {};
+
+    auto* object = Object::create_empty(global_object);
+    object->set_prototype(global_object.object_prototype());
+
+    get_iterator_values(global_object, iterable, [&](Value iterator_value) {
+        if (vm.exception())
+            return IterationDecision::Break;
+        if (!iterator_value.is_object()) {
+            vm.throw_exception<TypeError>(global_object, ErrorType::NotAnObject, String::formatted("Iterator value {}", iterator_value.to_string_without_side_effects()));
+            return IterationDecision::Break;
+        }
+        auto key = iterator_value.as_object().get(0).value_or(js_undefined());
+        if (vm.exception())
+            return IterationDecision::Break;
+        auto value = iterator_value.as_object().get(1).value_or(js_undefined());
+        if (vm.exception())
+            return IterationDecision::Break;
+        auto property_key = key.to_property_key(global_object);
+        if (vm.exception())
+            return IterationDecision::Break;
+        object->define_property(property_key, value);
+        return vm.exception() ? IterationDecision::Break : IterationDecision::Continue;
```

Comment:
This seems unnecessarily "smart" at the cost of readability :)

```suggestion
        if (vm.exception())
            return IterationDecision::Break;
        return IterationDecision::Continue;
```

Reply:
fixed!